core/controller/src/main/scala/whisk/core/loadBalancer/DistributedLoadBalancerData.scala

```diff
@@ -17,69 +17,102 @@
 
 package whisk.core.loadBalancer
 
+import akka.actor.Actor
+import akka.actor.ActorRef
 import akka.actor.ActorSystem
+import akka.actor.Props
 import akka.util.Timeout
-import akka.pattern.ask
-import whisk.common.Logging
-import whisk.core.entity.{ActivationId, UUID}
-
-import scala.collection.concurrent.TrieMap
-import scala.concurrent.Future
+import scala.collection.mutable
 import scala.concurrent.duration._
+import whisk.common.Logging
+import whisk.core.entity.ActivationId
+import whisk.core.entity.InstanceId
+import whisk.core.entity.UUID
 
 /**
  * Encapsulates data used for loadbalancer and active-ack bookkeeping.
  *
  * Note: The state keeping is backed by distributed akka actors. All CRUDs operations are done on local values, thus
  * a stale value might be read.
  */
-class DistributedLoadBalancerData(implicit actorSystem: ActorSystem, logging: Logging) extends LoadBalancerData {
+class DistributedLoadBalancerData(instance: InstanceId, monitor: Option[ActorRef] = None)(
+  implicit actorSystem: ActorSystem,
+  logging: Logging)
+    extends LoadBalancerData {
 
   implicit val timeout = Timeout(5.seconds)
   implicit val executionContext = actorSystem.dispatcher
-  private val activationsById = TrieMap[ActivationId, ActivationEntry]()
+  private val overflowKey = "overflow"
+  private val activationsById = mutable.Map[ActivationId, ActivationEntry]()
```

Comment:
Why remove the `TrieMap` and make this a (non-concurrent) `Map`?

Reply:
I was going down the path of making lb an actor; but that can be a separate discussion, possibly.